### ISSUE TITLE
Table loading issue with more than 5,000 rows on version 1.11.1

### ISSUE THREAD

**Issue by sadafr** [wenzhixin/bootstrap-table#3167]
The solution provided in [https://github.com/wenzhixin/bootstrap-table/issues/2352](url) worked for me for bootstrap version 1.8.1. Now i have upgraded to version 1.11 and again facing same issue even with 5,000 records. Test with IE chrome and Firefox, all are taking too much time to load grid that is more than a minute or two for this number. Kindly suggest me a solution for this as my records might go more than 80,000 with 'All' option and I have serve side pagination.

In addition, guide me can we integrate https://clusterize.js.org/ with bootstrap table ??

Awaiting for your reply.

Thanks.

**Comment by djhvscf** [comment]
Yes, this is an urgent issue. We know that the main issue is in the initBody function where we are trying to improve the performance..

We will keep you posted about this urgent issue.

**Comment by sadafr** [comment]
Thanks for looking into this and hope to hear from you soon.

Also suggest me how can improve performance if we have all actions(search, sorting, pagination) are server side ?

**Comment by djhvscf** [comment]
We need improve this function https://github.com/wenzhixin/bootstrap-table/blob/develop/src/bootstrap-table.js#L1608, if any one can help us it would be great

We are already using [document.createDocumentFragment](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment) which supposes that it is more efficient than use the append function but the performance is very slow when we have more than 10.000 rows

**Comment by ibesora** [comment]
I'm willing to help but I don't really know if there are tests to validate the results.

On your [getting-started guide ](http://bootstrap-table.wenzhixin.net.cn/getting-started/) it says that in order to run tests you should use `grunt test` but that doesn't seem to exists. In fact there is only a task defined to build the docs and another one to build everything without passing any test. 

Looking at the Gruntfile.js history it seems the test task has never been present in any version of it and doing a folder-wide search it seems all the references of tests are in the bower.json file, where it's found in the ignore list, and in the getting-started files

**Comment by raku** [comment]
Are you ever display 10000 rows once? If not why to render them? Render only 10 / 50 / 100 or how many page size you have defined.

**Comment by sadafr** [comment]
any update @djhvscf  ?

**Comment by sadafr** [comment]
@wenzhixin  &  @djhvscf any solution ?

For all option, can we control scroll to load records in chunks ?  If someone tried this please share..

**Comment by djhvscf** [comment]
No updates yet. We need to rewrite the init function

**Comment by djhvscf** [comment]
Working on this performance issue https://github.com/wenzhixin/bootstrap-table/tree/feature/performance_issue

**Comment by sadafr** [comment]
Thanks @djhvscf for working on this issue. Please update me once the solution is ready.

**Comment by pchaganti** [comment]
👍

**Comment by wenzhixin** [comment]
Merged into develop branch.
